File: gsb/gsb-crm/src/main/java/com/gongsibao/crm/web/SysSalesmanListPart.java
package com.gongsibao.crm.web;

import java.util.ArrayList;
import java.util.List;

import org.netsharp.communication.ServiceFactory;
import org.netsharp.panda.commerce.ListPart;
import org.netsharp.util.StringManager;

import com.gongsibao.supplier.base.ISalesmanService;
import com.gongsibao.utils.SupplierSessionManager;

public class SysSalesmanListPart extends ListPart {

	
	ISalesmanService salesmanService = ServiceFactory.create(ISalesmanService.class);
	
	/**   
	 * @Title: setDisabled   
	 * @Description: TODO(设置停用/启用状态)   
	 * @param: @param salesmanId
	 * @param: @param state
	 * @param: @return      
	 * @return: boolean      
	 * @throws   
	 */
	public boolean setDisabled(Integer salesmanId, boolean state) {

		return salesmanService.setDisabled(salesmanId, state);
	}

	/**   
	 * @Title: setReceiving   
	 * @Description: TODO(设置接单状态)   
	 * @param: @param salesmanId
	 * @param: @param state
	 * @param: @return      
	 * @return: boolean      
	 * @throws   
	 */
	public boolean setReceiving(Integer salesmanId, boolean state) {

		return salesmanService.setReceiving(salesmanId, state);
	}

	@Override
	protected String getExtraFilter() {

		List<String> ss = new ArrayList<String>();

		if (this.pdatagrid.getLazy()) {

			String filter = getRequest("filter");
			if (!StringManager.isNullOrEmpty(filter)) {

				return null;
			}
			String id = this.getRequest("id");
			if (StringManager.isNullOrEmpty(id)) {

				ss.add("parentId=0 or parentId is null");
			} else {
				ss.add("parentId=" + id);
			}
		}

		// 过滤服务商ID
		String requestSupplierId = this.getRequest("supplierId");
		if (!StringManager.isNullOrEmpty(requestSupplierId)) {

			ss.add("supplierId=" + requestSupplierId);
		} else {

			Integer supplierId = SupplierSessionManager.getSupplierId();
			if (supplierId != null) {

				ss.add("supplierId=" + supplierId);
			}
		}
		return StringManager.join(" and ", ss);
	}
}

File: gsb/gsb-web/src/main/webapp/gsb/supplier/sys/organization/js/sys-salesman-list-part.js
System.Declare("com.gongsibao.crm.web");

com.gongsibao.crm.web.SysSalesmanListPart = org.netsharp.panda.commerce.ListPart.Extends( {

    ctor: function () {
        this.base();
    },
    doQuery : function(filter,queryParams) {
        

        var urls = this.getFilters(filter);
        var url = urls.join("&");
        url = System.Url.getUrl(url);
        url = encodeURI(url);
        this.resetUrl(url,queryParams);
        // this.reload();
    },
    onAdding : function() {

        var relationItem = this.relationItem;
        if(relationItem == null){

            IMessageBox.info("请选择部门");
            return false;
        }

        if(relationItem.children.length>0){

            IMessageBox.info("请选择末节点");
            return false;
        }
        return true;
    },
    getSelectedItem : function(obj) {
        if(obj==1){//obj 1代表新增,其他的不处理

          //新增不做选择校验
        }else {

            var items = this.getSelections();

            if (items.length == 0) {

                IMessageBox.info("您没有选择记录!");
                return null;
            } else if (items.length > 1) {

                IMessageBox.info("只能选择一条记录!");
                return null;
            } else {
                return items[0];
            }
        }




    },
    add: function() {

        if (!this.onAdding()) {
            return;
        }
        var fks = [];

        if (this.context.relationRole != null && this.relationItem != null) {

            fks.push(this.context.relationRole + ":"+ this.relationItem.id);
        }

        var selectedRow = this.getSelectedItem(1);
        if (selectedRow == null) {

            var supplierId = this.queryString('supplierId');
            if(supplierId){
                fks.push("supplierId:" + supplierId);
            }
            this.doAdd("fk=" + fks.join(";"));

        } else {

            var parentId = this.getSelectedItem().id;
            if (parentId != null && parentId != "") {
                fks.push("parentId:" + parentId);
            }

            var supplierId = this.queryString('supplierId');
            if(supplierId){
                fks.push("supplierId:" + supplierId);
            }

            this.doAdd("fk=" + fks.join(";"));
        }
    },
    addExtraParams:function(urls){// 进行扩展

        var supplierId = this.queryString('supplierId');
        if(supplierId){

            urls.push("supplierId=" + supplierId);
        }
    },
    disabledFormatter:function(value,row,index){
    	
		return '<input class="easyui-switchbutton" data-options="'
		+'checked:'+(!value)
		+',onText:\'启用\',offText:\'停用\','
		+'onChange:function(checked){controllerdepartments.setDisabled(\''+row.id+'\',!checked);}">';
    },
    setDisabled:function(salesmanId,state){
    	
		var me = this;
		this.invokeService("setDisabled", [salesmanId,state], function(data) {

			me.reload();
			IMessageBox.toast("操作成功！");
		});
    },
    receivingFormatter:function(value,row,index){

		return '<input class="easyui-switchbutton" data-options="'
		+'checked:'+value
		+',onText:\'是\',offText:\'否\','
		+'onChange:function(checked){controllerdepartments.setReceiving(\''+row.id+'\',checked);}">';
    },    
    setReceiving:function(salesmanId,state){
    	
		var me = this;
		this.invokeService("setReceiving", [salesmanId,state], function(data) {

			me.reload();
			IMessageBox.toast("操作成功！");
		});
    },  
	onLoadSuccess:function(data){
		
		$('.easyui-switchbutton').switchbutton();
	}
});


//重新调用查询
$(function() {

    controllerdepartments.query();
});